--- scripts/local.orig	2007-08-17 23:42:12.000000000 -0700
+++ scripts/local	2007-08-17 23:42:38.000000000 -0700
@@ -39,6 +39,11 @@
 		fi
 	fi
 
+	if [ ! -e "${ROOT}" ]; then
+		# Try mdadm in degraded mode in case some drive has failed.
+		/scripts/local-top/mdadm
+	fi
+
 	# We've given up, but we'll let the user fix matters if they can
 	while [ ! -e "${ROOT}" ]; do
 		echo "	Check root= bootarg cat /proc/cmdline"
--- scripts/local-top/mdadm.orig	2007-08-17 23:41:32.000000000 -0700
+++ scripts/local-top/mdadm	2007-08-17 23:32:19.000000000 -0700
@@ -16,6 +16,8 @@
 	echo "$PREREQ"
 }
 
+MD_DEGRADED_ARGS=
+
 case ${1:-} in
   prereqs)
     prereqs
@@ -23,6 +25,7 @@
     ;;
   from-udev)
     # yuk
+    MD_DEGRADED_ARGS=--no-degraded
     quiet=n
     log_begin_msg (){ echo "beginning: $*"; }
     log_success_msg (){ echo "success: $*"; }
@@ -48,7 +51,7 @@
   case "$quiet" in y*|Y*|1|t*|T*)
     return 1;;
   *)
-    return 0;;
+    return 1;;
   esac
 }
 
@@ -90,7 +93,6 @@
   MD_DEVS=all
 fi
 
-: "${MD_DEGRADED_ARGS:= --no-degraded}"
 
 if [ "$MD_DEVS" = all ]; then
   
